### PR TITLE
add getHorizontalID

### PR DESCRIPTION
**Pull request by Axionize**
Helper function to Blockface class to get HorizontalID of a blockface

### CODE REVIEW

**Comment by booky10** [comment]
Any reason for this being a static method?

**Comment by Axionize** [comment]
Apologies. This was largely copied from my Grim fork. Fixed.

File: api/src/main/java/com/github/retrooper/packetevents/protocol/world/BlockFace.java [hunk 1]
```diff
@@ -153,6 +153,32 @@ public BlockFace getCW() {
...
+    /** Returns int representing cardinal directions with south as the starting index at 0. UP and DOWN return -1
+     * <p>
+     * Used by Mojang for calculations involving mob AI, decorations, paintings, item frames, maps,
+     * structure generation, campfires, stairs, connecting blocks, and flowerbeds.
+     * Also used for getting Directions and Blockfaces as Rotations.
+     *
+     * @return int horizontal id
+     */
+    public int getHorizontalID(BlockFace facing) {
+        switch (facing) {
```
Comment (booky10) [inline api/src/main/java/com/github/retrooper/packetevents/protocol/world/BlockFace.java]: could you also remove the parameter and replace "facing" with "this"?

**Review by booky10 — CHANGES_REQUESTED** [review]

File: api/src/main/java/com/github/retrooper/packetevents/protocol/world/BlockFace.java (hunk 1 above)
Comment (Axionize) [reply api/src/main/java/com/github/retrooper/packetevents/protocol/world/BlockFace.java]: Done

**Review by booky10 — APPROVED** [review]

**Pull request merged**